lib/SPIRV/SPIRVToOCL.cpp

```diff
@@ -319,6 +323,41 @@ void SPIRVToOCL::visitCallSPIRVPipeBuiltin(CallInst *CI, Op OC) {
       &Attrs);
 }
 
+void SPIRVToOCL::visitCallSPIRVImageMediaBlockBuiltin(CallInst *CI, Op OC) {
+  AttributeList Attrs = CI->getCalledFunction()->getAttributes();
+  mutateCallInstOCL(
+      M, CI,
+      [=](CallInst *, std::vector<Value *> &Args) {
+        // Moving the first argument to the end.
+        std::rotate(Args.rbegin(), Args.rend() - 1, Args.rend());
+        Type *FuncType = CI->getType();
+        if (OC == OpSubgroupImageMediaBlockWriteINTEL) {
+          assert(Args.size() >= 4 && "Wrong media block write signature");
+          FuncType = Args.at(3)->getType(); // texel type
+        }
+        unsigned int BitWidth = FuncType->getScalarSizeInBits();
+        std::string FuncPostfix;
+        if (BitWidth == 8)
+          FuncPostfix = "_uc";
+        else if (BitWidth == 16)
+          FuncPostfix = "_us";
+        else if (BitWidth == 32)
+          FuncPostfix = "_ui";
+        else
+          assert(0 && "Unsupported texel type!");
+
+        if (FuncType->isVectorTy()) {
+          unsigned int NumEl = FuncType->getVectorNumElements();
```

Comment:
Please add a test for this piece of code: right now you don't have a test returning vector

Reply:
Done